keydata_test.go

```diff
@@ -362,30 +394,43 @@ func (s *keyDataTestBase) checkKeyDataJSONFromReaderAuthModeNone(c *C, r io.Read
 	s.checkKeyDataJSONDecodedAuthModeNone(c, j, creationParams, nmodels)
 }
 
-func (s *keyDataTestBase) checkKeyDataJSONDecodedAuthModePassphrase(c *C, j map[string]interface{}, creationParams *KeyParams, nmodels int, passphrase string, kdfOpts *KDFOptions) {
+func (s *keyDataTestBase) checkKeyDataJSONDecodedAuthModePassphrase(c *C, j map[string]interface{}, creationParams *KeyWithPassphraseParams, nmodels int, passphrase string, kdfOpts *KDFOptions) {
 	if kdfOpts == nil {
 		var def KDFOptions
 		kdfOpts = &def
 	}
 	var kdf testutil.MockKDF
 
-	costParams, err := kdfOpts.DeriveCostParams(0, &kdf)
+	costParams, err := kdfOpts.DeriveCostParams(32+16, &kdf)
 	c.Assert(err, IsNil)
 
-	s.checkKeyDataJSONCommon(c, j, creationParams, nmodels)
+	s.checkKeyDataJSONCommon(c, j, &creationParams.KeyParams, nmodels)
+
+	_, ok := j["kdf_alg"].(string)
+	c.Check(ok, testutil.IsTrue)
 
-	c.Check(j, Not(testutil.HasKey), "encrypted_payload")
+	version, ok := j["version"].(float64)
+	c.Check(ok, testutil.IsTrue)
+	c.Check(version, Equals, float64(2))
```

Comment:
Aha, ignore my earlier comment about these fields not being checked, but they should probably move to `checkKeyDataJSONCommon`.

Reply:
Moved.